packages/plugin-code/src/Code.ts

```diff
@@ -0,0 +1,111 @@
+import { CodeButtonDomObjectRenderer } from './CodeButtonDomObjectRenderer';
+import { CodeViewNode } from './CodeViewNode';
+import { CodeViewDomObjectRenderer } from './CodeViewDomObjectRenderer';
+import { VNode } from '../../core/src/VNodes/VNode';
+import { parseEditable } from '../../utils/src/configuration';
+import { DomLayoutEngine } from '../../plugin-dom-layout/src/DomLayoutEngine';
+import { isBlock } from '../../utils/src/isBlock';
+
+export class Code<T extends JWPluginConfig = JWPluginConfig> extends JWPlugin<T> {
+    codeView = new CodeViewNode();
+    readonly loadables: Loadables<Renderer & Layout> = {
+        renderers: [CodeButtonDomObjectRenderer, CodeViewDomObjectRenderer],
+        components: [
+            {
+                id: 'CodeButton',
+                render: async (): Promise<ActionableNode[]> => {
+                    const button = new ActionableNode({
+                        name: 'code',
+                        label: 'Toggle Code view',
+                        selected: (): boolean => this.active,
+                        modifiers: [new Attributes({ class: 'fas fa-code fa-fw' })],
+                    });
+                    return [button];
+                },
+            },
+            {
+                id: 'code',
+                render: async (): Promise<VNode[]> => {
+                    return [this.codeView];
+                },
+            },
+        ],
+        componentZones: [['CodeButton', ['actionables']]],
+    };
+    active = false;
+    async start(): Promise<void> {
+        await super.start();
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    async toggle(): Promise<void> {
+        if (this.active) {
+            this.deactivate();
+        } else {
+            this.activate();
+        }
+    }
+    async activate(): Promise<void> {
+        this.active = true;
+        const domLayoutEngine = this.editor.plugins.get(Layout).engines.dom as DomLayoutEngine;
+        const editable = domLayoutEngine.components.get('editable')[0];
+
+        // Update the view's contents.
+        const domEditable = domLayoutEngine.getDomNodes(editable)[0];
+        this.codeView.value = this._formatHtml(domEditable.cloneNode(true) as Element);
+
+        // Show the code view and hide the editable.
+        await this.editor.plugins.get(Layout).append('code', 'main');
+        await this.editor.execCommand('hide', { componentId: 'editable' });
+    }
+    async deactivate(): Promise<void> {
+        this.active = false;
+        const domLayoutEngine = this.editor.plugins.get(Layout).engines.dom as DomLayoutEngine;
+        const editable = domLayoutEngine.components.get('editable')[0];
+
+        // Parse the code view into the editable.
+        const fakeCodeView = document.createElement('div');
```

Comment:
How is that a "fake" code view ? Couldn't it be called a "container" or something along those lines ?